openmetadata-ui/src/main/resources/ui/src/components/Domain/AddDataProductModal/AddDataProductModal.component.tsx

```diff
@@ -0,0 +1,78 @@
+/*
+ *  Copyright 2023 Collate.
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+import { Button, Modal } from 'antd';
+import { useForm } from 'antd/lib/form/Form';
+import { CreateDataProduct } from 'generated/api/domains/createDataProduct';
+import { CreateDomain } from 'generated/api/domains/createDomain';
+import { Domain } from 'generated/entity/domains/domain';
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+import AddDomainForm from '../AddDomainForm/AddDomainForm.component';
+import { DomainFormType } from '../DomainPage.interface';
+
+interface AddDataProductModalProps {
+  open: boolean;
+  data?: Domain;
+  onCancel: () => void;
+  onSubmit: (data: CreateDomain | CreateDataProduct) => Promise<void>;
+}
+
+const AddDataProductModal = ({
+  open,
+  onSubmit,
+  onCancel,
+}: AddDataProductModalProps) => {
+  const { t } = useTranslation();
+  const [form] = useForm();
+
+  const handleFormSubmit = async (
```

Comment:
Don't think it's needed, since `AddDomainForm` is also component let's use that directly

Reply:
As discussed, for now we are keeping this modal and later create a separate modal just to handle forms